lib/manager/pub/readme.md

```diff
@@ -0,0 +1,286 @@
+## Overview
+
+#### Name of package manager
+
+[pub](https://pub.dartlang.org/)
+
+---
+
+#### What language does this support?
+
+Dart / Flutter SDK
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+No.
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+_pubspec.yaml_
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`(^|\/)pubspec\.yaml$`
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+Yes
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+YAML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Use YAML parser like [JS-YAML](https://github.com/nodeca/js-yaml)
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Yes, `dependencies` and `dev-dependencies`.
+
```

Comment:
there are also dependency_overrides if this is relevant here

Reply:
I think we can safely skip over them as they refer only to local paths